src/openms/source/ANALYSIS/OPENSWATH/OpenSwathScoring.cpp

```diff
@@ -164,33 +164,30 @@ namespace OpenMS
     getNormalized_library_intensities_(transitions, normalized_library_intensity);
 
     // find spectrum that is closest to the apex of the peak using binary search
-    OpenSwath::SpectrumPtr spectrum = fetchSpectrumSwath(used_swath_maps, imrmfeature->getRT(), add_up_spectra_, drift_lower, drift_upper);
+    std::vector<OpenSwath::SpectrumPtr> spectra = fetchSpectrumSwath(used_swath_maps, imrmfeature->getRT(), add_up_spectra_, drift_lower, drift_upper);
 
-    // calculate drift extraction width for current spectrum (with some extra for cross-correlation)
-    double drift_width = fabs(drift_upper - drift_lower);
-    double drift_lower_used = drift_lower - drift_width * im_drift_extra_pcnt_;
-    double drift_upper_used = drift_upper + drift_width * im_drift_extra_pcnt_;
+    // set the DIA parameters
+    double dia_extract_window_ = (double)diascoring.getParameters().getValue("dia_extraction_window");
+    bool dia_extraction_ppm_ = diascoring.getParameters().getValue("dia_extraction_unit") == "ppm";
```

Comment:
Not quite sure what the best way to cache this would be. It is already in the outer loop of the function. Should I chance these to public/protected variables? Should they be arguments for the function?

Reply:
the former, making use of `DefaultParamHandler::updateMembers()`.